src/regmod/composite_models/base.py

```diff
@@ -127,22 +131,29 @@ def add_offset(self, df: DataFrame, copy: bool = False) -> DataFrame:
         """
         df = df.copy() if copy else df
         if self.col_value in df:
-            df[self.data.col_offset] = link_funs[self.mtype](df[self.col_value])
+            df["offset"] = link_funs[self.mtype](df[self.col_value])
         return df
 
-    def get_data(self) -> DataFrame:
-        return self.data.df.copy()
+    def get_data(self) -> Optional[DataFrame]:
+        if self.df is not None:
+            return self.df.copy()
+        return None
 
     def set_data(self, df: DataFrame):
         if df.shape[0] == 0:
             raise ValueError("Attempt to use empty dataframe.")
-        self.data.attach_df(self.add_offset(df, copy=True))
+        self.df.attach_df(self.add_offset(df, copy=True))
```

Comment:
is `attach_df` a pandas method on dataframe?

Reply:
Thanks David! This is definitely wrong, will add a comment to this, when we work on `composite_model` we can solve this.